products_utils.ts

```diff
@@ -0,0 +1,110 @@
+import { getFromLocalStorage, setToLocalStorage } from './local_storage_utils'
+import ClickEvent = JQuery.ClickEvent
+import ChangeEvent = JQuery.ChangeEvent;
+
+let productsSpecial:Set<string> = new Set()
+let products:Array<string> = []
+
+const deleteProduct = (event: ClickEvent) => {
+    const selectedProduct = $(event.target)
+
+    selectedProduct.parents('.selected-product').remove()
+
+    const deleteProductName = String(selectedProduct.siblings('.selected-product__text').val())
+
+    productsSpecial.delete(deleteProductName)
+
+    const newArrayProduct:Array<string> = [...productsSpecial]
+
+    setToLocalStorage('products', newArrayProduct)
+
+    if (productsSpecial.size === 0) {
+        $('.navigation__buttons:eq(1)').attr('disabled', 'true')
+        $('.navigation__buttons:eq(2)').attr('disabled', 'true')
+    }
+
+    return productsSpecial
+}
+
+const editProduct = (event: ClickEvent) => {
+    const editProduct = $(event.target)
+
+    editProduct.siblings('.selected-product__text').css('text-align', 'left').attr('readOnly', 'false')
+
+    const deleteProductName = String(editProduct.siblings('.selected-product__text').val())
+
+    productsSpecial.delete(deleteProductName)
+
+    const newArrayProduct = [...productsSpecial]
+
+    setToLocalStorage('products', newArrayProduct)
+
+    return productsSpecial
+}
+
+const addLocalStorageToProducts = () => {
+    const products = getFromLocalStorage('products')
```

Comment:
I would like to improve this function to be generic.
It should look like this:
```
const products = getFromLocalStorage<Array<string>>('products')
```

you can use casting in TypeScript with `as` inside `getFromLocalStorage`